src/day5/main2.java

```diff
@@ -0,0 +1,209 @@
+package day5;
+
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class main2 {
```

Comment:
Good formatting, overall. Can try breaking the methods inside the class to smaller ones and move them to service classes. For starters, you can now try to make the methods smaller.